source/virtualBuffers/__init__.py

```diff
@@ -372,21 +374,22 @@ def loadBuffer(self):
 		self._loadProgressCallLater = wx.CallLater(1000, self._loadProgress)
 		threading.Thread(target=self._loadBuffer).start()
 
+	def fetchLabels(self):
+  		weblink=self.documentConstantIdentifier
+  		filename=customLabels.getFilenameFromElementDomain(weblink)
+   		labels=""
+   		config = configobj.ConfigObj(os.path.join(globalVars.appArgs.configPath, "webLabels\%s" % filename))
+   		for k,v in config.iteritems():
+   			labels+=k+":"+v+","
+   		return labels
+	
 	def _loadBuffer(self):
 		try:
-  			weblink=self.documentConstantIdentifier
-  			parsed_uri = urlparse( weblink )
-  		 	domain='{uri.netloc}'.format(uri=parsed_uri)
-   		 	domain=domain.replace('.','_')
-   		  	domain=domain.replace(':','_')
-   		  	domain=domain.replace('\\','_')
-   		  	filename=domain+'.ini'
-   		  	labels=""
-   		  	config = configobj.ConfigObj(os.path.join(globalVars.appArgs.configPath, "webLabels\%s" % filename))
-   		  	for k,v in config.iteritems():
-				labels+=k+":"+v+","
-			
-			self.VBufHandle=NVDAHelper.localLib.VBuf_createBuffer(self.rootNVDAObject.appModule.helperLocalBindingHandle,self.rootDocHandle,self.rootID,unicode(self.backendName),unicode(labels))
+			if (self.labelSupportOn):
+				labels=self.fetchLabels()
+				self.VBufHandle=NVDAHelper.localLib.VBuf_createBuffer(self.rootNVDAObject.appModule.helperLocalBindingHandle,self.rootDocHandle,self.rootID,unicode(self.backendName),unicode(labels))

```

Comment:
Rather than forcing labels to unicode here, it would be best that labels are unicode all of the way (I.e. they are unicode within the labels file itself if they are not already).